fluss-server/src/main/java/com/alibaba/fluss/server/kv/rocksdb/RocksDBMetricsManager.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.fluss.server.kv.rocksdb;
+
+import com.alibaba.fluss.utils.MapUtils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Global collector for RocksDB metrics that efficiently manages metric collection for thousands of
+ * RocksDB instances using a shared thread pool and batch processing.
+ */
+public class RocksDBMetricsManager {
+
+    private static final Logger LOG = LoggerFactory.getLogger(RocksDBMetricsManager.class);
+
+    /** Singleton instance. */
+    private static volatile RocksDBMetricsManager instance;
+
+    /** Update interval for metrics collection (in seconds). */
+    private static final long METRICS_UPDATE_INTERVAL_SECONDS = 5;
+
+    /** Number of threads for metrics collection. */
+    private static final int METRICS_COLLECTION_THREADS =
+            Math.max(1, Runtime.getRuntime().availableProcessors() / 4);
```

Comment:
IMO, single thread for rocksdb metrics updating is enough. A metric reporter shouldn't consume so many threads, otherwise, there must be too many metrics.